sql/upgrade/sl3.0.sql

```diff
@@ -1119,25 +1134,16 @@ INSERT INTO cr_report(chart_id, their_total, submitted, end_date, updated, enter
 -- The ID and matching post_date are entered in a temp table to pull the back into cr_report_line immediately after.
 -- Temp table will be dropped automatically at the end of the transaction.
 WITH cr_entry AS (
-SELECT cr.id::INT, cr.end_date, a.source, n.type, a.cleared::TIMESTAMP, a.amount::NUMERIC, a.transdate AS post_date, a.lsmb_entry_id
-    FROM :slschema.acc_trans a
-    JOIN :slschema.chart s ON chart_id=s.id
-    JOIN pg_temp.reconciliation__account_list() coa ON coa.accno=s.accno
-    JOIN account c on c.accno=s.accno
-    JOIN public.cr_report cr ON c.id = cr.chart_id
-    AND date_trunc('MONTH', a.transdate)::DATE <= date_trunc('MONTH', cr.end_date)::DATE
-    AND date_trunc('MONTH', a.cleared)::DATE   >= date_trunc('MONTH', cr.end_date)::DATE
-    AND ( a.cleared IS NOT NULL OR a.transdate > (SELECT MAX(cleared) FROM :slschema.acc_trans))
-    JOIN (
-        WITH types AS ( SELECT id,'AP' AS type FROM :slschema.ap
-                  UNION SELECT id,'AR'         FROM :slschema.ar
-                  UNION SELECT id,'GL'         FROM :slschema.gl)
-        SELECT DISTINCT ac.trans_id, types.type
-        FROM :slschema.acc_trans ac
-        JOIN types ON ac.trans_id = types.id
-        ORDER BY ac.trans_id
-    ) n ON n.trans_id = a.trans_id
-    ORDER BY post_date,cr.id,n.type,a.source ASC NULLS LAST,a.amount
+SELECT cr.id::INT, cr.end_date, a.source, a.type, a.cleared::TIMESTAMP, a.amount::NUMERIC, a.transdate AS post_date, a.lsmb_entry_id
```

Comment:
The equivalence between the old and new chunks needs some explanation in the commit message, because I'm not seeing it yet...

Reply:
This algorythm is O(n2) on a big table, thus we have to be carefull with it.
In order to reduce the time, I
- moved the computation of accno outside the loop and precomputed it at the beginning of the migration routine;
- same for the truncated dates, they were moved in an augmented `acc_trans` table and computed once;
- moved the type computation in `acc_trans` so it can show AR, AP or GL, like we do with `transactions`;
- the query order was modified to use the smallest tables in the beginning and the biggest at the end;
The query time dropped by an order of magnitude (10x)

Reply:
Thanks for the explanation, but it belongs in the commit message, so it is retained even if we have to move from GitHub to another provider some day. This explanation would have made a great commit message.